src/OpenRiaServices.Hosting.AspNetCore/Framework/AspNetCore/Serialization/ArrayPoolStream.cs

```diff
@@ -161,6 +162,60 @@ public BufferMemory GetBufferMemoryAndReset()
             return res;
         }
 
+        public ArraySegment<byte> GetRentedArrayAndClear()
```

Comment:
This method is copied from `BufferManagerStream.GetArrayAndClear` from WCF project with only minor changes.
I think it is a good idea to look at that method when reviewing.

The main difference is that we don't have any offset in this class